single_kernel_mongo/events/ldap.py

```diff
@@ -0,0 +1,118 @@
+        self.relation_name = ExternalRequirerRelations.LDAP.value
+        self.cert_relation_name = ExternalRequirerRelations.LDAP_CERT.value
+
+        super().__init__(parent=dependent, key=self.relation_name)
+
+        self.framework.observe(self.manager.ldap_requirer.on.ldap_ready, self._on_ldap_ready)
+        self.framework.observe(
+            self.manager.ldap_requirer.on.ldap_unavailable, self._on_ldap_unavailable
+        )
+        self.framework.observe(
+            self.manager.certificate_transfer.on.certificate_available,
+            self._on_certificate_available,
+        )
+        self.framework.observe(
+            self.manager.certificate_transfer.on.certificate_removed, self._on_certificate_removed
+        )
+        self.framework.observe(self.restart_if_ready_event, self._on_restart_if_ready)
+
+        self.framework.observe(
+            self.charm.on["ldap-peers"].relation_changed, self._on_restart_if_ready
+        )
+
+    def _on_ldap_ready(self, event: LdapReadyEvent):
+        """Handles the ops event that indicates that ldap relation is ready."""
+        action = "ldap-ready"
+        try:
+            self.manager.on_ldap_ready(event.relation)
+        except WaitingForLdapDataError as err:
+            self.charm.status_manager.set_and_share_status(WaitingStatus("Waiting for LDAP data."))
+            defer_event_with_info_log(logger, event, action, f"{err}")
+
+        except (DeferrableError, DeferrableFailedHookChecksError) as err:
+            defer_event_with_info_log(logger, event, action, f"{err}")
+        except LDAPSNotEnabledError:
+            self.charm.status_manager.set_and_share_status(
+                BlockedStatus("LDAPS not enabled on LDAP application.")
+            )
+        except NonDeferrableFailedHookChecksError as err:
+            logger.error(f"{err}")
+            self.charm.status_manager.set_and_share_status(BlockedStatus(err.args[0]))
+
+    def _on_ldap_unavailable(self, event: LdapUnavailableEvent):
+        """Handles the ops event that indicates that ldap relation is now unavailable."""
+        self.manager.on_ldap_unavailable()
+
+    def _on_certificate_available(self, event: CertificateAvailableEvent):
+        """Handles the ops event that indicates that ldap-certificates relation is ready."""
+        try:
+            self.manager.on_certificate_available(event.certificate, event.ca, event.chain)
+        except DeferrableFailedHookChecksError as err:
+            defer_event_with_info_log(logger, event, "ldap-cert-ready", f"{err}")
+        except NonDeferrableFailedHookChecksError as err:
+            logger.error(f"{err}")
+            self.charm.status_manager.set_and_share_status(BlockedStatus(err.args[0]))
+
+    def _on_certificate_removed(self, event: CertificateRemovedEvent):
+        """Handles the ops event that indicates that ldap-certificates relation is unavailable."""
+        self.manager.on_certificate_removed()
```

Comment:
Similar comment on manager methods naming, it is unclear from here what exactly does the manager do in this hook